### PR TITLE
✨  add lancet module

### PR DESCRIPTION
**Pull request by dmiller15**
# Pull Request Template

## Description

The Lancet module and associated tools.

Part of https://github.com/d3b-center/bixu-tracker/issues/1337

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] https://cavatica.sbgenomics.com/u/d3b-bixu/dev-canine-workflow/tasks/e6cae665-d08e-4544-911e-bed54b2840d5/#

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by sickler-alex — APPROVED** [review]

File: subworkflows/canine_lancet_module.cwl [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    in:
+      input_file: lancet/vcf 
+      output_filename:
+        valueFrom: $(inputs.input_file.nameroot).sorted.vcf
+      output_type:
+        valueFrom: "v"
+      fai:
+        source: indexed_reference_fasta
+        valueFrom: |
+          $(self.secondaryFiles.filter(function(e){ return e.nameext == '.fai' })[0])
```
Comment (migbro) [inline subworkflows/canine_lancet_module.cwl:81]: nifty!

File: tools/bcftools_reheader_sort_index.cwl [hunk 2]
```diff
@@ -0,0 +1,85 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: bcftools_reheader_sort_index
+doc: |
+  BCFTOOLS reheader, sort, and optionall index
```
Comment (migbro) [inline tools/bcftools_reheader_sort_index.cwl:5]:
```suggestion
  BCFTOOLS reheader, sort, and optional index
```

**Review by migbro — APPROVED** [review]
Just a few questions and comments, but looks good overall

File: subworkflows/canine_lancet_module.cwl (hunk 1 above)
Comment (dmiller15) [reply subworkflows/canine_lancet_module.cwl:81]: Yeah was surprised with this one worked without dying.

File: tools/bcftools_concat_sort_index.cwl [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  rm_dups:
+    type:
+      - 'null'
+      - type: enum
+        name: rm_dups
+        symbols: ["snps", "indels", "both", "all", "none"]
+    inputBinding:
+      prefix: "--rm-dups"
+      position: 2
+  remove_duplicates: { type: 'boolean?', inputBinding: { position: 2, prefix: "--remove-duplicates"}, doc: "Alias for --rm-dups none" }
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/bcftools_concat_sort_index.cwl:50]: I don't know why they do that these descriptions were just taken directly from the help.

File: tools/bcftools_reheader_sort_index.cwl (hunk 2 above)
Comment (dmiller15) [reply tools/bcftools_reheader_sort_index.cwl:5]: Nice

File: tools/bcftools_stats.cwl [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  - position: 0
+    shellQuote: false
+    valueFrom: >
+      bcftools stats 
+
+stdout: $(inputs.output_filename)
+
+inputs:
+  # Required Inputs
+  input_vcf: { type: 'File', inputBinding: { position: 9 }, doc: "VCF files to concat, sort, and optionally index" }
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/bcftools_stats.cwl:25]: I copy paste a lot.

**Pull request merged**